### ISSUE TITLE
Epoch jumps during training

### ISSUE THREAD

**Issue by Cli98** [zylo117/Yet-Another-EfficientDet-Pytorch#268]
Hi @zylo117 ,

I observed another issue during training. I firstly pull up-to-date code from your repo yesterday then train the model. There's no error when I train my model, but epoch jumps.

<img width="524" alt="image" src="https://user-images.githubusercontent.com/19630616/81320048-91765b80-905e-11ea-8e20-5ee1278d3f77.png">

The model trains epoch 1 with only 100 steps. Is this the expected behavior?

**Comment by zylo117** [comment]
can't see it on your picture. where is it exactly?
Epoch will change if batchsize is changed when you load the last weights.

**Comment by Cli98** [comment]
> can't see it on your picture. where is it exactly?
> Epoch will change if batchsize is changed when you load the last weights.

@zylo117 I did not reload my model, just call training command and keep it running.
As I said, epoch 0 uses **15504** steps to finish, but epoch 1 needs **only 100** steps. Why could that happen?
I did not reload my checkpoint and so batch_size keeps the same.

**Comment by zylo117** [comment]
looks like a minor bug, but it doesn't seems to affect the training process. I'll look into it.

**Comment by Cli98** [comment]
Thank you,

**Comment by zylo117** [comment]
It is probably caused by this line https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/train.py#L231.
When your loss is infinte or some exception is raised, the iter here will still counts.
https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/train.py#L210
